File: app/src/main/java/de/bitmacht/workingtitle36/view/RegularView.java
package de.bitmacht.workingtitle36.view;

import android.annotation.TargetApi;
import android.content.Context;
import android.os.Build;
import android.util.AttributeSet;
import android.widget.TextView;

import de.bitmacht.workingtitle36.R;
import de.bitmacht.workingtitle36.RegularModel;

public class RegularView extends BaseTransactionView {

    private TextView periodView;

    public RegularView(Context context, AttributeSet attrs) {
        super(context, attrs);
    }

    public RegularView(Context context, AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
    }

    @TargetApi(Build.VERSION_CODES.LOLLIPOP)
    public RegularView(Context context, AttributeSet attrs, int defStyleAttr, int defStyleRes) {
        super(context, attrs, defStyleAttr, defStyleRes);
    }

    @Override
    protected void onFinishInflate() {
        super.onFinishInflate();
        periodView = (TextView) findViewById(R.id.period);
    }

    public void setData(RegularModel regularModel) {
        setValue(regularModel.getValue());
        periodView.setText(getPeriodText(regularModel));
        setDescription(regularModel.description);
    }

    private CharSequence getPeriodText(RegularModel regularModel) {
        return String.valueOf(regularModel.periodType) + ":" + String.valueOf(regularModel.periodMultiplier);
    }
}

File: app/src/main/java/de/bitmacht/workingtitle36/view/RegularView.kt
/*
 * Copyright 2016 Kamil Sartys
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package de.bitmacht.workingtitle36.view

import android.annotation.TargetApi
import android.content.Context
import android.os.Build
import android.util.AttributeSet
import android.widget.TextView

import de.bitmacht.workingtitle36.R
import de.bitmacht.workingtitle36.RegularModel

class RegularView : BaseTransactionView {

    private lateinit var periodView: TextView

    constructor(context: Context, attrs: AttributeSet) : super(context, attrs)

    constructor(context: Context, attrs: AttributeSet, defStyleAttr: Int) : super(context, attrs, defStyleAttr)

    @TargetApi(Build.VERSION_CODES.LOLLIPOP)
    constructor(context: Context, attrs: AttributeSet, defStyleAttr: Int, defStyleRes: Int) : super(context, attrs, defStyleAttr, defStyleRes)

    override fun onFinishInflate() {
        super.onFinishInflate()
        periodView = findViewById(R.id.period) as TextView
    }

    fun setData(regularModel: RegularModel) {
        value = regularModel.value
        periodView.text = regularModel.getPeriodString(context)
        description = regularModel.description
        alpha = if (regularModel.isDisabled) 0.6f else 1.0f
    }
}
